### PR TITLE
chore(deps): bump github.com/argoproj/argo-events from 1.7.3 to 1.9.1

### PR DESCRIPTION
**Pull request by blkperl**
### Motivation

Add the necessary lint and generated files for https://github.com/argoproj/argo-workflows/pull/12505. Supersedes/Closes #12505

### Modifications

* updated `go.mod` to the latest `argo-events`
* fixed deprecated functions in `test/e2e/argo_server_test.go`
* ran `make pre-commit -B` for generated files

### Verification

None, existing tests should be sufficient

### CODE REVIEW

**Comment by blkperl** [comment]
The test failure looks like https://github.com/argoproj/argo-workflows/issues/12836

**Review by terrytangyuan — COMMENTED** [review]
Why not upgrading to 1.9.1?

**Comment by blkperl** [comment]
@terrytangyuan oops, sorry the title was wrong but the go.mod file was 1.9.1

File: test/e2e/argo_server_test.go
Comment (agilgur5) [inline test/e2e/argo_server_test.go:1]: For reference, this file and part of the `go.mod` are the only non-generated changes

**Review by agilgur5 — APPROVED** [review]
LGTM, thanks for taking care of the Events dep update

**Comment by agilgur5** [comment]
Re-ran tests due to #12836

**Comment by blkperl** [comment]
Thank you! @agilgur5 

Also the closes/fixes syntax is unnecessary for dependabot prs as it will automatically self close when it detects a dependency has been updated

**Comment by agilgur5** [comment]
> Also the closes/fixes syntax is unnecessary for dependabot prs as it will automatically self close when it detects a dependency has been updated

It will, but it's good to know _why_ it's no longer needed and the closes syntax makes an explicit closing back-link

**Pull request merged**